### DISCUSSION TITLE
ESP32 Support?

### DISCUSSION THREAD

**Discussion by askdesigners** [bxparks/AceButton#91] [General]
I'm wanting to use this lib with an ESP32, but I'm not using the Arduino IDE or Arduino library. Will it still work?

**Comment by bxparks** [comment]
How can I answer this question if you don't tell me what you are using instead of the Arduino IDE? How about you just try to use it, and you tell me if it works or not?

**Reply by askdesigners** [reply]
Wow friendly! 

I'm using VSCode with the official ESP32 extension, which I believe uses IDF under the hood. Helpful?

**Reply by askdesigners** [reply]
More specifically, I'm wondering if it would handle this method of binding to the pin. 
```
  gpio_set_direction(button_pin, GPIO_MODE_INPUT);
  gpio_pulldown_en(button_pin);
  gpio_set_intr_type(button_pin, GPIO_INTR_ANYEDGE);
```
